pkg/apis/serving/v1alpha1/revision_types.go

```diff
@@ -229,6 +229,12 @@ type RevisionSpec struct {
 	// +optional
 	Container corev1.Container `json:"container,omitempty"`
 
+	// ImagePullSecrets defines a list of secrets which can be used to authenticate
+	// against a private secure registry. Secret(s) must exist in the same namespace
+	// as a revision's pod.
+	// +optional
+	ImagePullSecrets []corev1.LocalObjectReference `json:"imagePullSecrets,omitempty"`
```

Comment:
Two other places this is needed (off the top of my head):
1. This needs to be incorporated into the tag-to-digest stuff in the Revision controller.
1. This needs to be copied over to the Image resource that the Revision controller creates ([here](https://github.com/knative/serving/blob/b59542c426044261c10ed3f96b18e6a5bcc1dc07/pkg/reconciler/v1alpha1/revision/resources/imagecache.go#L50))

Reply:
@mattmoor I checked the code and since `ImagePullSecrets` is per pod, then adding this logic to controller does not seem to be needed.  Controller config only relevant to a queue container, which becomes a part of revision deployment/pod. Please let me know wdyt.